src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/InputFallThrough.java

```diff
@@ -521,4 +521,14 @@ void tryResource() throws Exception {
         public void close() throws Exception {
         }
     }
+
+    void synchronizedStatement() {
+       switch (hashCode()) {
+           case 1:
+               synchronized (this) {
+                   break;
+               }
+           default:
+       }
+    }
```

Comment:
one test is not enough, please add test that do cause violations with `synchronized`.
and one more compicated test with `synchronized` nested in `if` for example or smth similar.

Reply:
done